src/app/services/session-storage.service.ts

```diff
@@ -20,15 +20,30 @@ export class SessionStorageService {
     else this._data = JSON.parse(dataRaw);
   }
 
+  /**
+   * Updates sessions storage
+   */
   updateSessionData(key: string, value: any): void {
     if (value != null) this._data[key] = value;
     else delete this._data[key];
 
     sessionStorage.setItem('fem-alert', JSON.stringify(this.serialize(this._data)));
   }
 
+  /**
+   * Clears session storage
+   */
+  clearSessionData(): void {
+    sessionStorage.clear();
+  }
```

Comment:
why don't you clear the `_data` property of the service and update the sessionStorage itself with that value? the point of the service was to have as minimal points in the application that use the sessionStorage itself and use the `_data` property to manage data that is needed during a session. with the current implementation the `_data` property and the sessionStorage could be possible synchronized.

Reply:
don't know how to do it since its a read only

Reply:
you can the `readonly` modifier, and do the following: `this._data = {}` to clear the session data, and then overwrite the sessionStorage itself by `sessionStorage.setItem('fem-alert', this.serialize(this._data));`

Reply:
you can [remove]? the readonly modifier?